src/test/java/edu/byu/cs/server/ServerTest.java

```diff
@@ -0,0 +1,148 @@
+
+class ServerTest {
+    private static TestServerFacade serverFacade;
+    private static Server server;
+
+    private static final MockEndpointProvider mockedMockProvider = spy(new MockEndpointProvider());
+
+    public static Stream<Arguments> getPathParamEndpoints() {
+        return Stream.of(
+                Arguments.of("GET", "commitAnalyticsGet", "/api/admin/analytics/commit", ":option"),
+                Arguments.of("GET", "honorCheckerZipGet", "/api/admin/honorChecker/zip", ":section"),
+                Arguments.of("GET", "submissionXGet", "/api/submission", ":phase"),
+                Arguments.of("GET", "latestSubmissionsGet", "/api/admin/submissions/latest", ":count"),
+                Arguments.of("GET", "studentSubmissionsGet", "/api/admin/submissions/student", ":netid")
+        );
+    }
+
+    public static Stream<Arguments> getEndpoints() {
+        return Stream.of(
+                Arguments.of("GET", "callbackGet", "/auth/callback"),
+                Arguments.of("GET", "loginGet", "/auth/login"),
+                Arguments.of("GET", "usersGet", "/api/admin/users"),
+                Arguments.of("GET", "testModeGet", "/api/admin/test_mode"),
+                Arguments.of("GET", "commitAnalyticsGet", "/api/admin/analytics/commit"),
+                Arguments.of("GET", "sectionsGet", "/api/admin/sections"),
+                Arguments.of("GET", "meGet", "/api/me"),
+                Arguments.of("GET", "getConfigAdmin", "/api/admin/config"),
+                Arguments.of("GET", "getConfigStudent", "/api/config"),
+                Arguments.of("GET", "updateCourseIdsUsingCanvasGet", "/api/admin/config/courseIds"),
+                Arguments.of("GET", "submitGet", "/api/submit"),
+                Arguments.of("GET", "latestSubmissionForMeGet", "/api/latest"),
+                Arguments.of("GET", "submissionXGet", "/api/submission"),
+                Arguments.of("GET", "latestSubmissionsGet", "/api/admin/submissions/latest"),
+                Arguments.of("GET", "submissionsActiveGet", "/api/admin/submissions/active"),
+                Arguments.of("GET", "repoHistoryAdminGet", "/api/admin/repo/history"),
+                Arguments.of("POST", "logoutPost", "/auth/logout"),
+                Arguments.of("POST", "updateLivePhases", "/api/admin/config/phases"),
+                Arguments.of("POST", "updateBannerMessage", "/api/admin/config/banner"),
+                Arguments.of("POST", "updateCourseIdsPost", "/api/admin/config/courseIds"),
+                Arguments.of("POST", "submitPost", "/api/submit"),
+                Arguments.of("POST", "adminRepoSubmitPost", "/api/admin/submit"),
+                Arguments.of("POST", "approveSubmissionPost", "/api/admin/submissions/approve"),
+                Arguments.of("POST", "submissionsReRunPost", "/api/admin/submissions/rerun")
+        );
+    }
+
+    // TODO figure out how to test PATCH calls... HttpURLConnection thinks it's an invalid method
```

Comment:
Sooooooo I don't remember why I started using PATCH calls. I remember thinking I had a good reason, but I didn't write it down and the more we deal with it, the sillier it seems. I'd be ok going through the endpoints and switching them over to more conventional PUT requests